test_wsgi_kerberos.py

```diff
@@ -120,6 +120,16 @@ def test_unauthorized(self):
         self.assertEqual(r.headers['www-authenticate'], 'Negotiate')
         self.assertEqual(r.headers['content-type'], 'text/plain')
 
+    def test_read_max_on_auth_fail(self):
+        '''
+        KerberosAuthMiddleware's ``read_max_on_auth_fail`` constructor param should
+        allow customizing reading of request bodies of unauthenticated requests.
+        '''
+        app = TestApp(KerberosAuthMiddleware(index, read_max_on_auth_fail=1))
+        resp = app.post('/', 'unread request body', expect_errors=True)
+        self.assertEqual(resp.status_int, 401)
+        self.assertGreater(resp.request.body_file.raw.remaining, 0)
```

Comment:
This test doesn't appear to work as intended - it passes regardless of `read_max_on_auth_fail` value. Not sure what the problem is, but if it's fixable, we should have 2 tests for `len(body) > read_max_on_auth_fail` and `len(body) < read_max_on_auth_fail` to catch this.

Reply:
Yikes, good catch, thanks! Dug through the WebTest API a bit more and it looks like the best way to create a test request with a body that we can check how much has been read after submitting the request is to use `webtest.TestRequest` to explicitly create the test request with a `body` param. After submitting the request, we can then call `tell()` on the request body file "input" object to get the number of bytes read. Done in the latest revision, which also adds tests for a bunch of different read_max values that exercise all the possible cases. Another look?